scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -228,7 +228,7 @@ func (km *KafkaManager) Infer(
 	}
 
 	ctx, span := km.tracer.Start(ctx, "Produce")
-	span.SetAttributes(attribute.String(util.RequestIdHeader, key))
+	span.SetAttributes(attribute.String(util.RequestIdHeader, requestId))
```

Comment:
do we want to differentiate between the span of the main and shadow pipeline? I am thinking that the shadow pipeline might be using a faster model perhaps?

Reply:
Yeah that's a good point, but in that case we need to set a new attribute for the resource name and that affects other places as well. @lc525 What do you think?

Reply:
There will be two separate paths in the tracing tree for shadow and main pipeline which have the same request id as an attribute. This will be followed up if we do need a different attribute key to differentiate those.

Reply:
Note that this logic is now applied to spans in gateway https://github.com/SeldonIO/seldon-core/pull/5275/commits/cc0e9a6d937a90bcbf347e84e6c9b8ab0a951d9b